sphinx_gallery/scrapers.py

```diff
@@ -110,6 +111,17 @@ def matplotlib_scraper(block, block_vars, gallery_conf, **kwargs):
     from matplotlib.animation import Animation
     image_path_iterator = block_vars['image_path_iterator']
     image_rsts = []
+    # Check for srcset hidpi images
+    srcset = gallery_conf.get('matplotlib_srcset', None)
+    hidpi = srcset is not None
+    if hidpi:
+        hidpi_mult_facs = []
+        for st in srcset:
+            if (len(st) > 0) and (st[-1] == 'x'):
+                hidpi_mult_facs += [float(st[:-1])]
+            else:
+                hidpi_mult_facs += [1]
```

Comment:
It is better to do this conversion in `_complete_gallery_conf` so that if it fails, it does so immediately before trying to build any examples. For example you can make `matplotlib_srcset` always a list of float in there, thereby translating `None` to `[1.]` or so.